packages/zksend/src/experimental/wallet/index.ts

```diff
@@ -0,0 +1,454 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Transaction } from '@mysten/sui/transactions';
+import { fromBase64, toBase64 } from '@mysten/sui/utils';
+import type {
+	StandardConnectFeature,
+	StandardConnectMethod,
+	StandardDisconnectFeature,
+	StandardDisconnectMethod,
+	StandardEventsFeature,
+	StandardEventsListeners,
+	StandardEventsOnMethod,
+	SuiChain,
+	SuiSignAndExecuteTransactionFeature,
+	SuiSignAndExecuteTransactionMethod,
+	SuiSignPersonalMessageFeature,
+	SuiSignPersonalMessageMethod,
+	SuiSignTransactionBlockFeature,
+	SuiSignTransactionBlockMethod,
+	SuiSignTransactionFeature,
+	SuiSignTransactionMethod,
+	Wallet,
+} from '@mysten/wallet-standard';
+import { getWallets, ReadonlyWalletAccount, SUI_MAINNET_CHAIN } from '@mysten/wallet-standard';
+import type { Emitter } from 'mitt';
+import mitt from 'mitt';
+
+import { DEFAULT_STASHED_ORIGIN, StashedPopup } from './channel/index.js';
+import type { StashedSupportedNetwork } from './types.js';
+
+const PACKAGE_VERSION = 'v1';
+
+type WalletEventsMap = {
+	[E in keyof StandardEventsListeners]: Parameters<StandardEventsListeners[E]>[0];
+};
+
+const STASHED_SESSION_KEY = 'stashed:session';
+
+let embeddedIframe: HTMLIFrameElement;
+
+let walletStatusCheckEnabled = false;
+let stashedWalletInstance: StashedWallet | null = null;
+let stashedWalletOrigin: string;
+let walletStatusIntervalId: NodeJS.Timeout | null = null;
+
+export const STASHED_WALLET_NAME = 'Stashed' as const;
+type StashedAccount = { address: string; publicKey?: string };
+
+const getStashedSession = (): { accounts: StashedAccount[]; token: string } => {
+	const { accounts = [], token } = JSON.parse(localStorage.getItem(STASHED_SESSION_KEY) || '{}');
+	return { accounts, token };
+};
+
+const getPostMessagePayload = () => {
```

Comment:
We should similarly create a standard for the iframe messaging and ensure that we have schemas + validation + version it.